components/layout/flow_ref.rs

```diff
@@ -92,3 +134,72 @@ impl Clone for FlowRef {
         }
     }
 }
+
+impl WeakFlowRef {
+    /// Upgrades a WeakFlowRef to a FlowRef.
+    pub fn upgrade(&self) -> Option<FlowRef> {
+        unsafe {
+            let object = flow::base(&**self);
```

Comment:
Can you copy the comment from `arc.rs`:

        // We use a CAS loop to increment the strong count instead of a fetch_add because once the
        // count hits 0 is must never be above 0.